flow/raster_cache.h

```diff
@@ -88,6 +92,31 @@ struct RasterCacheMetrics {
 
 class RasterCache {
  public:
+  struct Context {
+    GrDirectContext* gr_context;
+    SkColorSpace* dst_color_space;
+    const SkMatrix& matrix;
+    const SkRect& logical_rect;
+    const char* flow_type;
+    bool checkerboard;
+  };
+
+  static std::unique_ptr<RasterCacheResult> Rasterize(
+      const RasterCache::Context& context,
+      const std::function<void(SkCanvas*)>& draw_function);
+
+  static bool IsDisplayListWorthRasterizing(
```

Comment:
Please remove any type-specific code from RasterCache. This class should become only concerned about RasterCacheKeys/RasterCacheEntries and leave the policies to the type-specific classes like the LayerItem, DLItem and SkPictureItem.